File: primetest.go
package main

import "fmt"

func main() {
    limit := 1000000
    count := 0

    for n := 2; n < limit; n++ {
        isPrime := true
        for i := 2; i*i <= n; i++ {
            if n%i == 0 {
                isPrime = false
            }
        }
        if isPrime {
            count++
        }
    }

    fmt.Printf("Primes found: %d\n", count)
}

File: primetest.rs
fn main() {
    let limit = 1000000;
    let mut count = 0;

    for n in 2..limit {
        let mut is_prime = true;
        let mut i = 2;
        while i * i <= n {
            if n % i == 0 {
                is_prime = false;
            }
            i += 1;
        }
        if is_prime {
            count += 1;
        }
    }

    println!("Primes found: {}", count);
}
